### ISSUE TITLE
Steam deletes entire game cache if opened twice during update and one is canceled during unpacking. 

### ISSUE THREAD

**Issue by NolanSyKinsley** [ValveSoftware/steam-for-linux#3677]
This is separate from #3671.

I opened steam twice on accident without realizing it and it was downloading an update. I canceled one of the download windows and let the other continue, the one I canceled popped up with an "unpacking"  status window and I canceled that as well, as it was an incomplete download. The other continued as expected. I have a feeling it was the cancelling during this unpacking process that caused the issue.

When it finished it nolonger had my username/password in it, so I put them in and logged in (Steamguard did not proc and it did not send me an e-mail, so it partly recognized my device as authorized). To my surprise my entire steam library was listed as uninstalled. I restarted steam to be sure and it still listed them as not installed, so I went to where they should have been installed (~/.local/share/steam/steamapps/common, or ~/.steam/steam/steamapps/common) and sure enough, the entire common folder was missing/deleted, not just unrecognized, but entirely removed.

I did no moving of the game cache or anything like that, I used the default location steam recommended (I have / and /home as two separate partitions).

Currently running manjaro(arch) 64 bit with xfce4 wm. If you need any more information let me know.

**Comment by xDShot** [comment]
I just wanted to add this issue today. I got the same issue.

Yesterday Steam downloaded update. It installed, and my games were kept.

However, in this morning when I launched steam, 'first-time login' window were popped up. It 'forgot' my log/pass so I re-typed them. And my games were removed :(

What is very interesting, I use Manjaro x64 with Xfce too! Is it possible that this is a maintainers' problem?

**Comment by jazztickets** [comment]
Has happened to me twice now. One thing you could do in case it happens again is just create a hardlink to your common directory. Something like "cp -rl ~/.local/share/steam/steamapps/common .common_backup". That way if steam deletes it you'll still have the files without any additional hard drive space being used. Although i guess you'd still have to keep recreating the hardlink when a game gets updated with new files.

**Comment by garciabruno** [comment]
I can confirm this, I was logged out after an update occurred. My entire game library was missing, although it still exists in ~/.local/share/Steam/steamapps

**Comment by neilsh** [comment]
I've run into this twice, having to re-install my entire game cache each time is really unfortunate.

**Comment by Skiski** [comment]
I has the same problem than Garciabruno happened to me at least 4 or 5 times after updates. When I launch Steam, it asks me to log in, the when I get to my library nothing was installed and even the non-steam games links were removed. Restarting Steam solves the problem. Nothing was deleted.

**Comment by NolanSyKinsley** [comment]
Ok, this just happened AGAIN today. I turn my computer on and lo and behold, steam is asking for my username/password, even though I have it set, I entered it, it came up with a quick display of "updating information" and BAM all my games wiped. I even tried to backup my game files beforehand, but the backup tool kept locking up.

Seriously, this is the worst bug I could imagine for a gaming platform, I don't want to have to keep going through 3 days of downloading and dealing with slow internet every 5 days because steam keeps wiping the folders on my hard drive, under NO circumstances should steam be deleting the ENTIRE game cache.

Just to make this perfectly clear, this is not steam refusing to recognize games on my hard drive, this is steam DELETING 130gb of games and removing ANY trace of them.

**Comment by Noiwex** [comment]
I had this just yesterday. My whole steamapps directory was removed, including my uncommitted changes in SVN folder. This is just embarrassing.

**Comment by slagerfin** [comment]
this happened to me yesterday

**Comment by Noiwex** [comment]
This happened to my friend too, just this morning. Valve, please fix it.

**Comment by SunDwarf** [comment]
Happened to me too.

**Comment by descrip** [comment]
This also happened to me.

**Comment by lbaile200** [comment]
I have had this issue as well.  Exact same conditions as posted.  I'm severely bandwidth capped where I am and I lost my entire library (close to 30 games).  Luckily I had backups of a few of my favorite games, but this is still inconvenient.

**Comment by Noiwex** [comment]
You could try having steamapps as symlink to somewhere else, I think.

**Comment by kon14** [comment]
Do count me in as well, this has already happened more than 3 times to me. The distro and setup shouldn't really make any difference but here it is anyway: I'm running Gentoo
Steps to reproduce: 
The steam client starts, then tries to update, fails unpacking on first attempt (I believe there was a "you have to be online", while I was online, comment last time), I then try to launch steam, updates fine, asks for user login, steam guard partially recognizes my computer, all library deleted :/

**Comment by bjwest** [comment]
**What are you people doing?**
After, I assume, a Steam update today, (I had to reboot due to a system update and didn't stick around for the reboot) I come back from getting a cup of coffee and I see the Steam login screen asking for my password.  Remembering seeing this thread in my inbox, I decide to look around before logging in.  I see a newly created ~/Steam directory containing 'appcache' and 'logs' sub directories.  I also see a newly created ~/.steam directory with the folllowing structure:
```
drwxrwxr-x   2 bjwest bjwest  4096 Jan 31 07:25 .
drwxr-xr-x 120 bjwest bjwest 12288 Jan 31 07:25 ..
lrwxrwxrwx   1 bjwest bjwest    25 Jan 31 07:25 bin -> /home/bjwest/.steam/bin32
lrwxrwxrwx   1 bjwest bjwest    43 Jan 31 07:25 bin32 -> /home/bjwest/.local/share/Steam/ubuntu12_32
lrwxrwxrwx   1 bjwest bjwest    43 Jan 31 07:25 bin64 -> /home/bjwest/.local/share/Steam/ubuntu12_64
-rw-rw-r--   1 bjwest bjwest  2550 Jan 31 07:25 registry.vdf
lrwxrwxrwx   1 bjwest bjwest    31 Jan 31 07:25 root -> /home/bjwest/.local/share/Steam
lrwxrwxrwx   1 bjwest bjwest    39 Jan 31 07:25 sdk32 -> /home/bjwest/.local/share/Steam/linux32
lrwxrwxrwx   1 bjwest bjwest    39 Jan 31 07:25 sdk64 -> /home/bjwest/.local/share/Steam/linux64
lrwxrwxrwx   1 bjwest bjwest    31 Jan 31 07:25 steam -> /home/bjwest/.local/share/Steam
-rwxr-xr-x   1 bjwest bjwest  8860 May 22  2014 steam_install_agreement.txt
-rw-rw-r--   1 bjwest bjwest     5 Jan 31 07:25 steam.pid
prw-------   1 bjwest bjwest     0 Jan 29 09:30 steam.pipe
```
Needless to say, I'm copying my entire ~./local/share/Steam directory to a backup drive before I log in.  I have no faith Steam will not screw something up as soon as I enter my password.

Why do I now have three directories for Steam, and why is one of them a muddled up symlink mess of the original ~/.local/share/Steam?  

Do you guys have any quality control going on over there, or do you just push out any old crap to us without testing?  With reports of home directories being wiped, I'm getting very leery of any update, and am wondering if I should forget about Steam for a while and return to my PS3.

You seriously need to get this under control.  Steam is not a stand-alone console.  Most of us run it on our primary computers, and you guys are sending out updates that are wiping out personal data.

**Comment by bjwest** [comment]
After backing up my original Steam directory, I entered my password and here's what happened.

I had no games installed, so selected to install 'Don't Starve' because it's small and wouldn't take too long.  Steam asked me where I wanted to install my SteamLibs directory.  Fearing what would happen, I installed a blank USB drive and chose that.  It complained it needed an empty directory (lost+found is in the root), so I created a directory SteamLibrary and selected that.  Steam created a steamapps subdirectory and created a libraryfolders.vdf file in there.  It then continued to install Don't Starve in ~/Seam/steamapps/common.

All my games were still in ~/.local/share/Steam/SteamApps/common, so I copied one of the games into ~/Steam/steamapps/common to see if Steam would pick it up. It didn't imediatly, so I exited Steam and restarted it.  It asked me for my password AGAIN, then low and behold all my games were there.

I unmounted the USB drive, restarted Steam again with no complaints.  Whatever this library directory is for, it's not needed.  I also deleted the dont_starve and Civ V directories from the ~/Steam/steamapps/common directory, restarted Steam and there was no change.  It's using the old common directory where all the games are stored.

My only conclusion is that there is absolutely no quality assurance going on in the Steam development department, and they are pushing out incomplete, buggy and sometimes extremely dangerous code, and we are left wondering if our systems will be completely destroyed in the process.

**tl:dr**  You may not need to reinstall all your games.  Just shut down Steam and restart it a couple of times and it *may* pick them back up.

**Comment by kspangsege** [comment]
This happened to me today.

**Comment by kspangsege** [comment]
Please see https://github.com/ValveSoftware/steam-for-linux/issues/3722 for details.

**Comment by alexpaknix** [comment]
I have a similar situation.
ticket 4756-WKLX-3055 from 23rd Feb 2015

Support recommended this issue https://support.steampowered.com/kb_article.php?ref=3134-TIAL-4638&l=russian

_steam --reset_
this action removes the installed games, setting and authorization data

**Comment by NolanSyKinsley** [comment]
Tested latest linux client, cannot reproduce original bug, nor is the bug present while using steam --reset anymore. It looks like it has been fixed, so I am closing this ticket.

**Comment by gogromat** [comment]
Just happened to me (March 8 2016) on Linux Debian (Linux 3.16.0-4-amd64 x86_64).
- Steam updated
- Forced me to re-login
- All games were missing in Steam - Steam asked to choose location for games to download.
- ```ls ~/.local/share/Steam/steamapps/common/``` showed that they are all intact.
- Found this post.
- Cancelled to choose new location for games
- Rebooted the machine
- Loaded Steam again
- All games are present again

**Comment by NolanSyKinsley** [comment]
@gogromat That was not the original issue here. The original issue was full deletion of the files, not just them missing from the list in steam. The entire game cache folder was being wiped.

If you are experiencing a new issue with them vanishing from your list on update but still being present on your drive then I would look for that issue, and if it is not present, open a new one.

**Comment by gogromat** [comment]
@NolanSyKinsley Thanks man, I couldn't find the same exact issue where there is no delete.
I was reading comments by garciabruno, Skiski and bjwest in this thread - they seem to have exactly the same issue as me, that's why I posted here.

**Comment by Anty-Lemon** [comment]
This occurred to me today. All games are completely messing from my SSD

**Comment by Noiwex** [comment]
Gabe pls

**Comment by NolanSyKinsley** [comment]
@Anty-Lemon A bug like this one, but not this bug, did happen to me as well the other day. Check your root steam folder for a .save folder, your game cache may very well still exist in that folder, it was where mine was.

In this original bug report they game files were completely wiped, not moved to a .save folder, so it is a different issue if that is what happened to you as well.

**Comment by Anty-Lemon** [comment]
There is no `.save` directory. Additionally, ~20 gigs of space were freed in `/home`.

**Comment by NolanSyKinsley** [comment]
Ah, then it may be a similar bug, but this issue was fixed over 2 years ago. It may be a regression or a new bug entirely. I would open a new issue with full system specs and such. Might link to this closed issue just in case.

**Comment by Anty-Lemon** [comment]
I'm skeptical, considering no commit was pushed to fix this bug, and your description of it matches my experience to a tee. I'm more liable to believe there's a race condition.

**Comment by na1pir** [comment]
I can confirm this! Lost all my games when on vacation with very pure wifi access.  so no gaming for me for entire weak.

**Comment by josealvim** [comment]
It just happened to me, it's the 3rd time already, so it's quite reproducible.

**Comment by NolanSyKinsley** [comment]
This bug just reared it's ugly head again
